client/src/app/pages/applications/components/application-detail-drawer/fact-detail-modal/fact-code-snip-viewer.tsx

```diff
@@ -0,0 +1,22 @@
+import * as React from "react";
+import { CodeEditor, Language } from "@patternfly/react-code-editor";
+import { Fact } from "@app/api/models";
+
+export interface IFactCodeSnipViewerProps {
+  fact: Fact;
+}
+
+export const FactCodeSnipViewer: React.FC<IFactCodeSnipViewerProps> = ({
+  fact,
+}) => {
+  return (
+    <CodeEditor
+      isReadOnly
+      isDarkTheme
+      isLineNumbersVisible
+      language={Language.json}
+      height="450px"
+      code={JSON.stringify(fact.data)}
```

Comment:
The design calls for displaying this in YAML rather than JSON right? Since we already have the `js-yaml` package installed we could `import yaml from 'js-yaml';` and then use `yaml.dump(fact.data)` below instead of `JSON.stringify(fact.data)`, and use `Language.yaml` instead of `Language.json`. I think that should be sufficient, although we may want to wrap the `yaml.dump` in a try-catch by lifting this code out into a variable. I suppose the API always gives us JSON and all JSON can be stringified to YAML so maybe that's fine.

Reply:
Ah, looking at [the docs](https://www.npmjs.com/package/js-yaml) we can avoid this ever throwing an exception by using the `skipInvalid` option:
```ts
code={yaml.dump(fact.data, { skipInvalid: true })}
```